src/application-base/ApplicationBase.jsx

```diff
@@ -106,19 +107,23 @@ const ApplicationBase = ({
         translationsLoadingPlaceholder={translationsLoadingPlaceholder}
         locale={locale}
       >
-        <ActiveBreakpointProvider>
-          <ApplicationErrorBoundary>
+        <ApplicationIntlProvider>
+          <ActiveBreakpointProvider>
             <NavigationPromptCheckpoint
               onPromptChange={(registeredPrompts) => {
                 registeredPromptsRef.current = registeredPrompts;
               }}
             >
               <ApplicationLoadingOverlayProvider>
-                {children}
+                <ApplicationErrorBoundary>
+                  <Suspense fallback={<ApplicationLoadingOverlay isOpen />}>
+                    {children}
+                  </Suspense>
+                </ApplicationErrorBoundary>
               </ApplicationLoadingOverlayProvider>
             </NavigationPromptCheckpoint>
-          </ApplicationErrorBoundary>
-        </ActiveBreakpointProvider>
+          </ActiveBreakpointProvider>
+        </ApplicationIntlProvider>
       </Base>
```

Comment:
Any changes we might want a top-level error boundary in case an error occurs before loading their content?

Reply:
My thinking was that we'd want to keep as much mounted as possible in the case of a child error. I guess we could move it up immediately under Base?